src/legacy/ui/public/chrome/directives/header_global_nav/components/header_badge.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiBetaBadge } from '@elastic/eui';
+import React, { Component } from 'react';
+import * as Rx from 'rxjs';
+
+interface Props {
+  badge$: Rx.Observable<ChromeBadge | undefined>;
+}
+
+interface State {
+  badge: ChromeBadge | undefined;
+}
+
+import { ChromeBadge } from '../../../../../../../core/public';
+
+export class HeaderBadge extends Component<Props, State> {
+  private subscription?: Rx.Subscription;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = { badge: undefined };
+  }
+
+  public componentDidMount() {
+    this.subscribe();
+  }
+
+  public componentDidUpdate(prevProps: Props) {
+    if (prevProps.badge$ === this.props.badge$) {
+      return;
+    }
+
+    this.unsubscribe();
+    this.subscribe();
+  }
+
+  public componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  public render() {
+    if (this.state.badge == null) {
+      return null;
+    }
+
+    return (
+      <EuiBetaBadge
```

Comment:
We've also decided to reduce it's footprint by using an icon instead of the full text like so:

<img width="240" alt="Screen Shot 2019-04-09 at 13 04 43 PM" src="https://user-images.githubusercontent.com/549577/55825614-53fa7400-5ad4-11e9-9ea2-fbe0004d4184.png">

But I have to get the glasses icon into EUI as it doesn't exist yet.

Reply:
Awesome, thanks so much @cchaos!

Reply:
https://github.com/elastic/kibana/pull/34824 is in Master with the glasses icon

Reply:
That was 10x faster than I anticipated, thanks @cchaos!